src/library/autodj/autodjprocessor.cpp

```diff
@@ -1674,6 +1669,18 @@ double AutoDJProcessor::getFadeTime() {
     DeckAttributes* pLeftDeck = m_decks[0];
     DeckAttributes* pRightDeck = m_decks[1];
 
+    // Calculate the BPM for tracks in both decks
+    TrackPointer leftTrack = pLeftDeck->getLoadedTrack();
+    TrackPointer rightTrack = pRightDeck->getLoadedTrack();
+
+    const double leftTrackBPM = leftTrack->getBpm();
+    const double rightTrackBPM = rightTrack->getBpm();
```

Comment:
This gets an average BPM for the whole track so it still assumes a constant tempo. AutoDJProcessor should not be concerned with the tempo for this feature; it only needs the length of time that the last beats of the track occupy (multiplied by rateRatio). This information must come from the Beats object.

Reply:
Interesting. Thanks. I'll work on that.

Reply:
Hope in commit  [33f83a4](https://github.com/mixxxdj/mixxx/pull/2746/commits/33f83a4ed49b2401a1e6d51599dce568cb2f780d) I'm moving in the right direction?